src/os/Transaction.h

```diff
@@ -153,6 +153,10 @@ class Transaction {
     OP_COLL_SET_BITS = 42, // cid, bits
 
     OP_MERGE_COLLECTION = 43, // cid, destination
+			      //
+    OP_LOG_SETKEYS = 44, // cid, attrset
+    OP_LOG_RMKEYS = 45,  // cid, keyset
+    OP_LOG_RMKEYRANGE = 46,  // cid, oid, firstkey, lastkey
```

Comment:
This extends the `os::Transaction` interface used by `BlueStore`, `MemStore`, `KStore`, `AlienStore` and also upper layers of both `crimson-osd` and `ceph-osd` . As this broad interface crosses the boundaries of single OSD, backward compatibility across replica set becomes an issue – I don't see a code handling it. All the long-taking deprecation rules (n-2) would apply which makes this place costly to change.

I talked today with @Matan-B and @cyx1231st, so it's clear for me it's just for experimenting within `SeaStore`. My point is the current way is pretty confusing as it touches the broad interface without even ifdefs.

Let's find a better way.

Reply:
I agree we should find a better way before introducing common OPs.
I'll note that after discussing with @athanatos, it seems that we are not actually sharing the new OPs on the wire (to replicas). Instead it is only used by each OSD to be committed to the store locally (`PG::log_operation` called in `PG::handle_rep_op`).
I'll keep working on this, will ping if something changed. Thank you!